app/main/util/search_utils.py

```diff
@@ -0,0 +1,174 @@
+    pagination = get_pagination(page, page_count)
+    return total_records, pagination
+
+
+def get_filtered_list(to_filter, filter):
+    """Filters a list based on the contents of another list"""
+    return [field for field in to_filter if field not in filter or []]
+
+
+def get_all_fields_excluding(open_search, index_name, exclude_fields=None):
+    """Retrieve all fields from the index and exclude certain fields"""
+    mappings = open_search.indices.get_mapping(index=index_name)
+    all_fields = list(mappings[index_name]["mappings"]["properties"].keys())
+    filtered_fields = get_filtered_list(all_fields, exclude_fields)
+
+    return filtered_fields
+
+
+def get_query_aggregations():
+    """Returns the aggregations segment of the DSL query"""
+    return {
+        "aggs": {
+            "aggregate_by_transferring_body": {
+                "terms": {"field": "transferring_body_id.keyword"},
+                "aggs": {
+                    "top_transferring_body_hits": {
+                        "top_hits": {
+                            "size": 1,
+                            "_source": ["transferring_body"],
+                        }
+                    }
+                },
+            }
+        },
+    }
+
+
+def get_query_multi_match(query, search_area, open_search, _id):
+    """Returns the multi-match segment of the DSL query"""
+
+    fields_record = ["file_name", "file_path", "content"]
+    fields_metadata = get_all_fields_excluding(
+        open_search, "documents", fields_record
+    )
+    fields = ["*"]
+
+    if search_area == "metadata":
+        fields = fields_metadata
+    elif search_area == "record":
+        fields = fields_record
+
+    must_clauses = [
+        {
+            "multi_match": {
+                "query": query,
+                "fields": fields,
+                "operator": "AND",
+                "fuzziness": "AUTO",
+                "lenient": True,
+            }
+        }
+    ]
+
+    filter_clauses = (
+        [{"term": {"transferring_body_id.keyword": _id}}] if _id else []
+    )
+
+    return {
+        "query": {
+            "bool": {
+                "must": must_clauses,
+                "filter": filter_clauses,
+            }
+        },
+    }
+
+
+def get_query_highlighting():
+    """Returns the highlight segment of the DSL query"""
+    return {
+        "highlight": {
+            "pre_tags": ["<mark>"],
+            "post_tags": ["</mark>"],
+            "fields": {
+                "*": {},
+            },
+        },
+    }
+
+
+def get_query_sorting():
+    """Returns the sort segment of the DSL query"""
+    sorting_query = build_sorting_orders_open_search(request.args)
+    return {
+        "sort": sorting_query,
+    }
+
+
+def get_query_source_rules():
+    """Returns the _source segment of the DSL query"""
+    return {"_source": {"exclude": ["*.keyword"]}}
+
+
+def build_dsl_query(query, search_area, open_search, _id=None):
```

Comment:
can we rename _id to transferring_body_id as there is noway to know it's meant to be that, and some context explaining that this function can be used to search across all bodies, or just the one if body is provided